lib-multisrc/cloudrecess/src/eu/kanade/tachiyomi/multisrc/cloudrecess/CloudRecess.kt

```diff
@@ -0,0 +1,294 @@
+                Log.e("$name: Filters", e.stackTraceToString())
+            }
+            fetchFilterAttempts++
+        }
+    }
+
+    protected open fun filtersRequest() = GET("$baseUrl/manga", headers)
+
+    protected open fun parseFilters(response: Response) {
+        val document = response.asJsoup()
+        typeList = document.select("select[name=type] > option").map {
+            it.ownText() to it.attr("value")
+        }
+        statusList = document.select("select[name=status] > option").map {
+            it.ownText() to it.attr("value")
+        }
+        genreList = document.select("div.grid > div.flex:has(> input[name=genre[]])").mapNotNull {
+            val label = it.selectFirst("label")?.ownText()
+                ?: return@mapNotNull null
+            val value = it.selectFirst("input")?.attr("value")
+                ?: return@mapNotNull null
+
+            label to value
+        }
+    }
+
+    override fun getFilterList(): FilterList {
+        CoroutineScope(Dispatchers.IO).launch { fetchFilters() }
+
+        val filters = mutableListOf<Filter<*>>()
+
+        if (typeList.isNotEmpty()) {
+            filters.add(TypeFilter(typeList))
+        }
+        if (statusList.isNotEmpty()) {
+            filters.add(StatusFilter(statusList))
+        }
+        if (genreList.isNotEmpty()) {
+            filters.add(GenreFilter(genreList))
+        }
+        if (filters.size < 3) {
+            filters.add(0, Filter.Header("Press 'reset' to load more filters"))
+        }
+
+        return FilterList(filters)
+    }
+
+    // details
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        val genres = mutableListOf<String>()
+        with(document.selectFirst("main > section > div")!!) {
+            thumbnail_url = selectFirst("div.relative img")?.imgAttr()
+            title = selectFirst("div.flex > h1, div.flex > h2")!!.ownText()
+            genres.addAll(select("div.flex > a.inline-block").eachText())
+            selectFirst("div:has(> p#description)")?.run {
+                select("#show-more").remove()
+                description = text()
+            }
+        }
+        document.selectFirst("div#buttons + div.hidden, div:has(> div#buttons) + div.flex")?.run {
+            status = getInfo("Status").parseStatus()
+            artist = getInfo("Artist") ?: getInfo("المؤلف")
+            author = getInfo("Author") ?: getInfo("الرسام")
+            (getInfo("Type") ?: getInfo("النوع"))?.also { genres.add(0, it) }
+        }
+        genre = genres.joinToString()
+    }
+
+    protected open fun String?.parseStatus(): Int {
+        this ?: return SManga.UNKNOWN
+
+        return when {
+            listOf("ongoing", "مستمر").any { contains(it, true) } -> SManga.ONGOING
+            listOf("cancelled", "متوقف").any { contains(it, true) } -> SManga.CANCELLED
+            listOf("Completed", "مكتمل").any { contains(it, true) } -> SManga.COMPLETED
```

Comment:
Since `contains` is used as case insensitive it should be lower case here

```suggestion
            listOf("completed", "مكتمل").any { contains(it, true) } -> SManga.COMPLETED
```